Improve explanation of CDDL and registration policy customization (#303)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

{{fig-signed-statement-cddl}} illustrates a normative CDDL definition (see {{-CDDL}}) for of the protected header and unprotected header of Signed Statements and Receipts.

This definition specifies the minimally required keys and values as mandatory.
Most keys and their respective values for the unprotected header, protected header, and the claims embedded therein are optional for Signed Statements and Receipts.
This design maximizes interoperability with these default requirements for all use cases, while permitting implementers to customize requirements for a specific use case through Registration Policy.

> For customizing requirements in Signed Statements and Receipts, a Registration Policy MAY define keys and/or values that are mandatory for that Transparency Service instance. Such a Transparency Service instance MUST reject any Signed Statements or Receipts that are invalid according to its current Registration Policy.

Reviewer: ```suggestion
For customizing requirements in Signed Statements and Receipts, a Registration Policy MAY define keys and/or values that are mandatory for that Transparency Service instance.
Such a Transparency Service instance MUST reject any Signed Statements or Receipts that are invalid according to its current Registration Policy.
```

{{fig-signed-statement-cddl}} illustrates a normative CDDL definition (see {{-CDDL}}) for of the protected header and unprotected header of Signed Statements and Receipts.

This definition specifies the minimally required keys and values as mandatory.
Most keys and their respective values for the unprotected header, protected header, and the claims embedded therein are optional for Signed Statements and Receipts.
This design maximizes interoperability with these default requirements for all use cases, while permitting implementers to customize requirements for a specific use case through Registration Policy.
For customizing requirements in Signed Statements and Receipts, a Registration Policy MAY define keys and/or values that are mandatory for that Transparency Service instance.

> Such a Transparency Service instance MUST reject any Signed Statements or Receipts that are invalid according to its current Registration Policy.

Reviewer: ```suggestion
Such a Transparency Service instance MUST reject any invalid Signed Statements or Receipts according to its current Registration Policy.
```

> This definition specifies the minimally required keys and values as mandatory.
> Most keys and their respective values for the unprotected header, protected header, and the claims embedded therein are optional for Signed Statements and Receipts.
> This design maximizes interoperability with these default requirements for all use cases, while permitting implementers to customize requirements for a specific use case through Registration Policy.
> For customizing requirements in Signed Statements and Receipts, a Registration Policy MAY define keys and/or values that are mandatory for that Transparency Service instance.
> Such a Transparency Service instance MUST reject any Signed Statements or Receipts that are invalid according to its current Registration Policy.

Reviewer: ```suggestion
This definition specifies the minimal mandatory labels.
Implementation-specific Registration Policies may define additional mandatory labels.
A Transparency Service implementation MUST reject registering Signed Statements that do not meet their current Registration Policy requirements.
Each implementation SHOULD provide details for their registration policies through documentation or discovery APIs.
```

Author: This proposal is terser and more to the point, I prefer this over my original edit, I take this suggestion.

Author: > I struggle with this section as it defines examples, yet there's only one minimal example for x.509. Another arbitrary example where label `foo` must be defined, with a value of `bar1` and `bar2` would be a combative example.

Also I was fine with the edit you proposed, but I am not sure how this part of this specification, with or without my edit, would have a contradiction in terms for `foo` versus `bar1` and `bar2`. If we follow the base architecture requirement, none are required.

I do like your simpler version, but I feel minimizing the use of precision using architecture and CDDL terminology is what makes this property less obvious, not more. Am I missing something?

Reviewer: The foo/bar proposal was the beginning of a proposal to add different examples. The heading says Signed Statement Example**s**, yet we only have one, with X.509. 
Since we don't seem ready to "endorse" a different identity type, I was thinking we might show some additional headers that MAY be required by another TS.

All that said, I'm fine with https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/pull/303/commits/123c8fca8a1f244bde8603a68d16b8dd4c51d465, and have LGTMd it.

Author: I can consider adding other examples tracked by a new issue, but do we have other workable examples as an alternative to X.509?

{{fig-signed-statement-cddl}} illustrates a normative CDDL definition (see {{-CDDL}}) for of the protected header and unprotected header of Signed Statements and Receipts.

> This definition specifies the minimally required keys and values as mandatory. Most keys and their respective values for the unprotected header, protected header, and the claims embedded therein are optional for Signed Statements and Receipts.

Reviewer: ```suggestion
This definition specifies the minimally required keys and values as mandatory.
Most keys and their respective values for the unprotected header, protected header, and the claims embedded therein are optional for Signed Statements and Receipts.
```
